latex2mmlc/src/parse.rs

```diff
@@ -504,10 +504,17 @@ impl<'source> Parser<'source> {
                 Node::MultiLetterIdent(StrReference::new(start, end))
             }
             Token::Text => {
-                self.check_lbrace()?;
-                // Read the text.
-                let text = self.parse_text_group()?;
-                Node::Text(text)
+                self.l.record_whitespace = true;
+                let node = self.parse_single_token()?.as_node(&self.arena);
+                let start = self.buffer.end();
+                extract_letters(&self.arena, &mut self.buffer, node)?;
+                let end = self.buffer.end();
+                self.l.record_whitespace = false;
+                // Discard any whitespace tokens that are still stored in self.peek_token.
+                if matches!(self.peek_token, Token::NonBreakingSpace) {
+                    self.next_token();
+                }
```

Comment:
This shouldn't discard legitimate `~` tokens. So, I'll probably need to introduce a new token.